### PR TITLE
Fixed StreamInfoDecoder to adapt to layout of XINFO response. Possible solution for #2818

### PR DESCRIPTION
**Pull request by fawitte**
Adds functionality for decoding StreamInfo with changing order of fields in response of XINFO.
It's a possible solution for #2818 and should also work if number or order of fields in XINFO response changes in the future.

### CODE REVIEW

**Comment by mrniko** [comment]
Please exclude RedissonTest from commit

**Comment by fawitte** [comment]
I reverted the refactoring commit.

Nikita Koksharov <notifications@github.com> schrieb am Fr., 5. Juni 2020,
18:07:

File: redisson/src/main/java/org/redisson/client/protocol/decoder/StreamInfoDecoder.java [hunk 1]
```diff
@@ -38,29 +38,70 @@
 
     @Override
     public StreamInfo<Object, Object> decode(List<Object> parts, State state) {
+        StreamInfoWrapper fields = new StreamInfoWrapper(parts);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/client/protocol/decoder/StreamInfoDecoder.java:41]: We don't need any wrapper here. List could be converted to map:
```java
        Map<String, Object> map = IntStream.range(0, parts.size()).boxed().collect(
                Collectors.groupingBy(e -> e / 2, Collectors.mapping(e -> parts.get(e), Collectors.toList())))
                .values().stream().collect(Collectors.toMap(e -> e.get(0), e -> e.get(1)));
```

Comment (fawitte) [reply redisson/src/main/java/org/redisson/client/protocol/decoder/StreamInfoDecoder.java:41]: I added the wrapper for mental support only, but this change is a very interesting solution.

Comment (fawitte) [reply redisson/src/main/java/org/redisson/client/protocol/decoder/StreamInfoDecoder.java]: I added the requested changes to the proposed solution.

**Comment by mrniko** [comment]
Thanks for the contribution!

**Pull request merged**